kfp/components/_components.py

```diff
@@ -547,8 +547,16 @@ def expand_command_part(arg) -> Union[str, List[str], None]:
         elif isinstance(arg, IfPlaceholder):
             arg = arg.if_structure
             condition_result = expand_command_part(arg.condition)
-            from distutils.util import strtobool
-            condition_result_bool = condition_result and strtobool(
+            # Replacement for distutils.util.strtobool (removed in Python 3.12)
+            def _strtobool(val):
+                val_lower = str(val).lower()
+                if val_lower in ('y', 'yes', 't', 'true', 'on', '1'):
+                    return True
+                elif val_lower in ('n', 'no', 'f', 'false', 'off', '0'):
+                    return False
+                else:
+                    raise ValueError(f"invalid truth value {val!r}")
+            condition_result_bool = condition_result and _strtobool(
```

Comment:
nit - You use the `strtobool` function here and in `_data_passing.py`. You could define once and import it.